jdbc/src/main/java/nextstep/jdbc/JdbcTemplate.java

```diff
@@ -14,4 +21,73 @@ public class JdbcTemplate {
     public JdbcTemplate(final DataSource dataSource) {
         this.dataSource = dataSource;
     }
+
+    public void update(final String sql, final Object... objects) {
+        executePreparedStatement(sql, pstmt -> {
+            setParams(pstmt, objects);
+            pstmt.executeUpdate();
+            return null;
+        });
+    }
+
+    public <T> List<T> query(final String sql, final RowMapper<T> rowMapper) {
+        return executePreparedStatement(sql, pstmt -> executeQuery(pstmt, rowMapper));
+    }
+
+    public <T> T queryForObject(final String sql, final RowMapper<T> rowMapper, final Object... objects) {
+        return executePreparedStatement(sql, pstmt -> {
+            setParams(pstmt, objects);
+            final var results = executeQuery(pstmt, rowMapper);
+            return nullableSingleResult(results);
+        });
+    }
+
+    private <T> T executePreparedStatement(final String sql, QueryExecutor<T> executor) {
```

Comment:
```suggestion
    private <T> T executePreparedStatement(final String sql, final QueryExecutor<T> executor) {
```

final 키워드가 빠진 것 같아요!